### PR TITLE
Add unit tests for Vocabulary elements

### PR DESCRIPTION
**Pull request by hamzanouali**
**Description**

Unit testing vocabulary elements.

**Checklist:**
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by HAMZABoxer98** [comment]
Why I am getting 

> **All checks have failed**

?!

File: tests/unit/elements/ImageView.spec.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+describe('ImageView.vue', () => {
+  const SOURCE = 'source demo';
+  const ALTERNATE_TEXT = 'hello';
+  const TITLE = 'title demo';
+  const PRIMARY_DIMENSION_HEIGHT = 'height';
+  const PRIMARY_DIMENSION_WIDTH = 'width';
+  const IS_CENTERED = true;
+  const IS_HOVERABLE = true;
+  const SIZE = 'mega';
+  const SLOT = '<strong>hello world</strong>';
```
Comment (dhruvkb) [inline tests/unit/elements/ImageView.spec.js]: Capitalisation here as well.

```html
<strong>Hello World!</strong>
```

**Comment by dhruvkb** [comment]
Do this 

```
git reset c154546
git push -f origin hamza-nouali-branch
```

**Comment by HAMZABoxer98** [comment]
@dhruvkb , I lost some files !!, I lost the component itself.

**Comment by dhruvkb** [comment]
@HAMZABoxer98 there are several [stages of `reset`](https://www.atlassian.com/git/tutorials/undoing-changes/git-reset).

**Comment by HAMZABoxer98** [comment]
The file has been removed, I couldn't find it in the previous commits or even my windows Trash.

**Comment by HAMZABoxer98** [comment]
@dhruvkb , I recovered it again 😄.
`git reflog` will show all the commits even the deleted ones.

File: tests/unit/elements/Button.spec.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            propsData: {
+            }
+        }
+    })
+
+    function getWrapper() {
+        return shallowMount(Button, defaultConfig)
+    }
+
+    it('renders addons\'s slot correctly', () => {
```
Comment (dhruvkb) [inline tests/unit/elements/Button.spec.js]: Change to 'addon slot'

File: tests/unit/elements/Button.spec.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+describe('Button.vue', () => {
+    const ROUNDNESS = 'complete'
+    const COLOR = 'red'
+    const ICON = 'vote-yea'
+    const SIZE = 'mega'
+    const INDICATION = 'positive'
+    const IS_INVERTIBLE = true
+    const IS_CALL_TO_ACTION = true
+    const SLOT = '<strong class="test-my-slot">Hello World</strong>'
+    let defaultConfig = null
```
Comment (dhruvkb) [inline tests/unit/elements/Button.spec.js]: Leave newline separating constants from `defaultConfig`

File: tests/unit/elements/Button.spec.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+    beforeEach(() => {
+        defaultConfig = {
+            propsData: {
+            }
```
Comment (dhruvkb) [inline tests/unit/elements/Button.spec.js]: Add `slots: {}` to `defaultConfig`

File: tests/unit/elements/Button.spec.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+    it('renders addons\'s slot correctly', () => {
+        defaultConfig.slots = {
+            addons: SLOT
+        }
+        const wrapper = getWrapper()
+        expect(wrapper.find('.addons').html()).toContain(SLOT)
+    })
+
+    it('renders content\'s slot correctly', () => {
```
Comment (dhruvkb) [inline tests/unit/elements/Button.spec.js]: Change to 'content slot'

File: tests/unit/elements/Button.spec.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    it('renders the correct invertion', () => {
+        defaultConfig.propsData.isInverted = IS_INVERTIBLE
+        const wrapper = getWrapper()
+        expect(wrapper.find('.button').classes()).toContain('inverted')
+    })
+
+    it('renders the correct indication', () => {
+        defaultConfig.propsData.indication = INDICATION
+        const wrapper = getWrapper()
+        expect(wrapper.find('.button').classes()).toContain('positive-indicating')
```
Comment (dhruvkb) [inline tests/unit/elements/Button.spec.js]: Use `${INDICATION}-indicating` as done for color

File: tests/unit/elements/Button.spec.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        expect(wrapper.find(`*[icon*="${ICON}"]`).exists()).toBe(true)
+    })
+
+    it('renders the correct color', () => {
+        defaultConfig.propsData.color = COLOR
+        const wrapper = getWrapper()
+        expect(wrapper.find('.button').classes()).toContain(`${COLOR}-colored`)
+    })
+
+    it('renders the correct invertion', () => {
```
Comment (dhruvkb) [inline tests/unit/elements/Button.spec.js]: The spelling is 'inversion'

File: tests/unit/elements/Button.spec.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+        defaultConfig = {
+            propsData: {
+            }
+        }
+    })
+
+    function getWrapper() {
+        return shallowMount(Button, defaultConfig)
+    }
+
```
Comment (dhruvkb) [inline tests/unit/elements/Button.spec.js]: Order tests based on the order of mixins and props in the component

File: tests/unit/elements/Button.spec.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+        expect(wrapper.find('.button').classes()).toContain('positive-indicating')
+    })
+
+    it('renders the correct scale/size', () => {
+        defaultConfig.propsData.size = SIZE
+        const wrapper = getWrapper()
+        expect(wrapper.find('.button').classes()).toContain('mega-sized')
+    })
+
+    it('renders the correct roundedness', () => {
```
Comment (dhruvkb) [inline tests/unit/elements/Button.spec.js]: The spelling is 'roundness'

File: tests/unit/elements/Button.spec.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
+import { shallowMount } from '@vue/test-utils'
+
+import Button from '@/elements/Button/Button'
+
+describe('Button.vue', () => {
+    const ROUNDNESS = 'complete'
```
Comment (dhruvkb) [inline tests/unit/elements/Button.spec.js]: Order of constants should be the in the order of tests
The order of tests is explained below

File: tests/unit/elements/ImageView.spec.js [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+    const wrapper = getWrapper()
+
+    expect(wrapper.find(`img[alt="${ALTERNATE_TEXT}"]`).exists()).toBe(true)
+  })
+
+  it('renders the correct title', () => {
+    defaultConfig.propsData.title = TITLE
+    defaultConfig.propsData.isHoverable = IS_HOVERABLE
+    const wrapper = getWrapper()
+
```
Comment (dhruvkb) [inline tests/unit/elements/ImageView.spec.js]: Remove all these newlines or alternatively leave newlines in all the tests
Consistency is important

File: tests/unit/elements/Button.spec.js [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+        }
+    })
+
+    function getWrapper() {
+        return shallowMount(Button, defaultConfig)
+    }
+
+    it('renders addons\'s slot correctly', () => {
+        defaultConfig.slots = {
+            addons: SLOT
```
Comment (dhruvkb) [inline tests/unit/elements/Button.spec.js]: Populate slots as `defaultConfig.slots.addons = SLOT`, just as done in the case of other props

**Comment by dhruvkb** [comment]
Could you go through the feedback and mark the feedback you incorporated as 'resolved'?

File: tests/unit/elements/Button.spec.js (hunk 12 above)
Comment (HAMZABoxer98) [reply tests/unit/elements/Button.spec.js]: Solved.

File: tests/unit/elements/ImageView.spec.js (hunk 11 above)
Comment (HAMZABoxer98) [reply tests/unit/elements/ImageView.spec.js]: Solved.

File: tests/unit/elements/Button.spec.js (hunk 10 above)
Comment (HAMZABoxer98) [reply tests/unit/elements/Button.spec.js]: Solved.

File: tests/unit/elements/Button.spec.js (hunk 8 above)
Comment (HAMZABoxer98) [reply tests/unit/elements/Button.spec.js]: Solved.

File: tests/unit/elements/Button.spec.js [hunk 13]
```diff
@@ -0,0 +1,89 @@
+import { shallowMount } from '@vue/test-utils'
```
Comment (dhruvkb) [inline tests/unit/elements/Button.spec.js]: This file is indented using 4 spaces. It should be 2.

File: tests/unit/elements/ImageView.spec.js [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+    }
+  })
+
+  function getWrapper() {
+    return shallowMount(ImageView, defaultConfig)
+  }
+
+  it('renders the correct image for the given source', () => {
+    const wrapper = getWrapper()
+
```
Comment (dhruvkb) [inline tests/unit/elements/ImageView.spec.js]: Remove newlines as newlines have not been left in above files.

File: tests/unit/elements/Rating.spec.js [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+      .toBe(1)
+    expect(wrapper.findAll(`*[icon*="${ICON_SET[2]}"]`).length)
+      .toBe(1)
+  })
+
+  it('single select', () => {
+    defaultConfig.propsData.isSingleSelect = IS_SINGLE_SELECTED
+    const wrapper = mount(Rating, defaultConfig)
+
+    wrapper.findAll('.icon.unit').at(2)
```
Comment (dhruvkb) [inline tests/unit/elements/Rating.spec.js]: If sequential functions don't fit on one line, put all of them on individual lines

```js
wrapper
  .findAll('icon.unit')
  .at(2)
  .trigger('click')
```

File: tests/unit/elements/SelectField.spec.js [hunk 16]
```diff
@@ -0,0 +1,94 @@
+import {
+  config,
+  shallowMount
+} from '@vue/test-utils'
```
Comment (dhruvkb) [inline tests/unit/elements/SelectField.spec.js]: This break is only done if the text does not fit entirely in one line.

**Comment by HAMZABoxer98** [comment]
> Sorry for being iterative, but found more code style changes.
I am happy to learn and make it a habit to follow the best practice 😄

**Comment by dhruvkb** [comment]
Appreciate your positive disposition to feedback!

**Comment by HAMZABoxer98** [comment]
@dhruvkb, why this PR still open? is there anything wrong in my code?

**Comment by dhruvkb** [comment]
No I just haven't had time because of Outreachy to re-review it and merge if everything works out.

**Comment by dhruvkb** [comment]
This PR adds tests for components that will eventually get deleted. It's unfortunate, but at this stage, merging this PR would only add more files to delete in later PRs.

**Pull request closed without merging**